lib/spack/spack/solver/asp.py

```diff
@@ -805,6 +811,12 @@ def on_model(model):
             print("Statistics:")
             pprint.pprint(self.control.statistics)
 
+        if result.unsolved_specs and setup.concretize_everything:
+            raise UnsatisfiableSpecError(
+                "Internal Spack error: the solver completed but produced specs"
+                " that do not satisfy the request. Rerun with 'spack -d'"
```

Comment:
Can we raise an `InternalConcretizerError`?

Reply:
`InternalConcretizerError` was originally specifically for cases where the solver could not find a solution (even with errors); I extended the exception hierarchy to add a `SolverError` as a subclass of `InternalConcretizerError` for this case.